### PR TITLE
Add SETA awarness to mozci #366

### PR DESCRIPTION
**Pull request by F3real**

### CODE REVIEW

**Comment by armenzg** [comment]
This is looking good.
We need a global dict which if not set we calculate the data.
If it's set, we then return the global dict.

We also need a function which given a buildername, we return the number of pushes for it.

For instance, in pulse_actions, we're using a MAX_REVISIONS variable [1] for every builder, however, we know that Windows XP builders have a different cadence.

A function probably called "get_SETA_info()".

[1] https://github.com/mozilla/pulse_actions/blob/master/pulse_actions/handlers/backfilling.py#L52

**Comment by F3real** [comment]
@armenzg r?

File: mozci/platforms.py [hunk 1]
```diff
@@ -94,6 +95,41 @@ def _process_data():
             BUILDERNAME_TO_TRIGGER[buildername.lower()] = values['triggered_by'][0]
 
 
+def get_SETA_interval_dict(force=False):
+
```
Comment (armenzg) [inline mozci/platforms.py:110]: Could you please add a docstring?

File: mozci/platforms.py [hunk 2]
```diff
@@ -94,6 +95,41 @@ def _process_data():
...
+        if not sched.startswith('tests-'):
+            continue
+
+        sched_str_list = sched.split('-')
+        try:
+            for buildername in values['downstream']:
+                seta = []
+                seta.append(int(sched_str_list[-2]))
+                seta.append(int(sched_str_list[-1]))
+                SETA_DICT[buildername] = seta
```
Comment (armenzg) [inline mozci/platforms.py:132]: Could you please add comments to explain what are we storing in here? (e.g. push and interval in minutes)

File: mozci/platforms.py [hunk 3]
```diff
@@ -94,6 +95,41 @@ def _process_data():
...
+        except (ValueError, IndexError):
+            continue
+    return SETA_DICT
+
+
+def get_SETA_info(buildername):
```
Comment (armenzg) [inline mozci/platforms.py:139]: Docstring and explain what we return.

File: mozci/platforms.py [hunk 4]
```diff
@@ -94,6 +95,41 @@ def _process_data():
...
+        sched_str_list = sched.split('-')
+        try:
+            for buildername in values['downstream']:
+                seta = []
+                seta.append(int(sched_str_list[-2]))
+                seta.append(int(sched_str_list[-1]))
+                SETA_DICT[buildername] = seta
+        except (ValueError, IndexError):
+            continue
+    return SETA_DICT
```
Comment (armenzg) [inline mozci/platforms.py:136]: A line blank above the return (visually  more clear).

**Comment by armenzg** [comment]
This looks good! Well done.
Once addressed the nits I can merge it and land it.

Once this is released we can change pulse_actions to use this (not that we to this in this quarter).

**Pull request closed without merging**